Missing <stdbool.h>

In HYPRE 2.33, the following files must `#include <stdbool.h>` in order to have macros `true` and `false` defined:

- src\distributed_ls\pilut\parilut.c
- src\distributed_ls\pilut\trifactor.c

Compilation fails with error C2065 on Visual Studio 2022.

--- a/src/distributed_ls/pilut/struct.h
+++ b/src/distributed_ls/pilut/struct.h
@@ -31,25 +31,17 @@
  */
 
 #ifndef __cplusplus  /* In C++, bool is built-in, no need to define it */
-#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L && __STDC_VERSION__ < 202311L
-/* C99 - C11: Use stdbool.h */
-#include <stdbool.h>
-#elif defined(__STDC_VERSION__) && __STDC_VERSION__ < 199901L
-/* Pre-C99: Define bool manually */
-#ifndef bool
-#ifdef Boolean
-   typedef Boolean bool;
-#else
-   typedef unsigned char bool;
-#endif
-#endif
-#ifndef true
-#define true 1
-#endif
-#ifndef false
-#define false 0
-#endif
-#endif
+   #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L && __STDC_VERSION__ < 202311L
+      /* C99 - C17: Use stdbool.h */
+      #include <stdbool.h>
+   #else
+      #ifndef true
+      #define true 1
+      #endif
+      #ifndef false
+      #define false 0
+      #endif
+   #endif
 #endif /* ifndef __cplusplus */
 
 /*************************************************************************

